File: Medium Problems/Leetcode 148.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* sortList(ListNode* head) {
        vector<int> arr;
        ListNode* p1 = head;
        while(p1 != nullptr){
            arr.push_back(p1->val);
            p1 = p1->next;
        }
        sort(arr.begin(),arr.end());
        p1 = head;
        for(int x : arr){
            p1->val = x;
            p1 = p1->next;
        }
        return head;
    }
};

File: Medium Problems/Leetcode 148.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        ls = []
        if not head or not head.next:
            return head
        tail = head
        while tail:
            ls.append(tail.val)
            tail = tail.next
        ls.sort()
        tail = head
        for i in ls:
            tail.val = i
            tail = tail.next
        return head